### PR TITLE
Implemented BST ability "Tame"

### PR DESCRIPTION
**Pull request by Wiggo32**
So, not much for reference online. I got most of my info from:
http://ffxiclopedia.wikia.com/wiki/Tame
and made 'best guess' as far as the values and percentages based on what is already in dsp and from what I "remember". Please let me know if there's anything I can/should do to make this better.

### CODE REVIEW

File: scripts/globals/abilities/tame.lua [hunk 1]
```diff
@@ -22,4 +22,48 @@ end;
...
+            tameChance = 99;
+        end
+
+    local charmChance = player:getCharmChance(target, false);
+    local pet = player:getPet();
+    local CE = target:getCE(player)
+    local mob = target:getID();
+
+    local canTame = true;
+    local party = player:getParty();
```
[the remark this answers is not in the document]
Comment (Wiggo32) [reply scripts/globals/abilities/tame.lua]: ahhh yes, you are correct. Should be easy.

**Comment by teschnei** [comment]
You could also check their hate list instead of the entire zone

**Comment by Wiggo32** [comment]
Yeah, looking at this... there are some loopholes/exploits that exist and I'm not sure I can do it properly without getting into the core.

**Comment by TeoTwawki** [comment]
```lua
    for _, players in pairs(mob:getEnmityList()) do
        DoStuff();
    end
```

Then you are getting the mobs hatelist instead of a party list or a zone's list, like teschnei suggested.

**Comment by Wiggo32** [comment]
Probably should close this... as it's only "as good as you can get it with current dsp configuration"
changed it up quite a bit on my server by including an "EFFECT_TAME" in the core which handles a lot of what was scripted up there. So far, that's what I feel is the best solution for MY server. Don't really want to hear how I did it wrong because I like it the way it is and it works well.

**Comment by teschnei** [comment]
I added core support for you to fix it a few weeks ago ;)

**Comment by Wiggo32** [comment]
How could you make that statement? I asked for very specific things to get this working right and they were largely ingored. You gave me "core support" to implement your suggestion but that's hardly 'all' it needed. As it stands now, dsp doesn't have the necessary core functions to get this retail accurate. You can get it 'close' but even in your own words that shouldn't be "good enough". If you guys do decide you want to get tame working the way it does on retail, I would suggest to start by adding a status EFFECT_TAME. That's going to clean up a lot of that code in those lua's and get rid of those silly char_vars; which is what dsp should be striving toward. But, what do I know? I'm just a poor sap contributor who's complaints and grievances can only be translated into 'anger' and 'rage'. Like I can't file a complaint without being 'angry'? I'm sorry but infliction on text is put there by the reader. I don't mean to direct this at you kj, or dump anything on you because out of all the "dsp op's" on discord you're the most mature, level headed, and respectable. But, my request for "core support" for tame went mostly ignored. I did chalk that up to life. Life happens and people are busy. I get it, understood. But, don't say that you gave me everything I needed when that is incorrect.

**Comment by TeoTwawki** [comment]
him: I added core support for you to fix it a few weeks ago

you: but that's hardly 'all' it needed. 

you: don't say that you gave me everything I needed when that is incorrect.

me: /confused

**Comment by teschnei** [comment]
Dang dude!

I added the first thing we were going to need (the additional data in the
enmity list exposed to Lua), I was waiting for you to use it before adding
anything else we needed, since I hadn't figured out exactly what we were
going to need yet.  Anyways, even though you deleted your repository, I
took what you had and finished most of it (and yes, it did need a bit more
core stuff, but development is an iterative process... I wanted to see
exactly what was left after you had used the enmity list changes I made for
you).

**Pull request closed without merging**